### PR TITLE
Profile pages routing...

### PR DESCRIPTION
**Pull request by mmmavis**
Related to #962

### CODE REVIEW

**Comment by mmmavis** [comment]
@xmatthewx this is ready for review

**Comment by xmatthewx** [comment]
Thanks @mmmavis. The routing is working great. I think you hit all [the requirements in the checklist](https://github.com/mozilla/network-pulse/issues/962#issue-326194183). Let's have the code reviewer double check, but I think it's good. 

I noticed a regression. 

When viewing someone else's profile, I see Project sub-headers and CTAs, when it should be empty.   
 - It only happens when the Profile has Projects under Published OR Created. (Everything works properly when the profile has both types or neither type.)
 - This profile ([id:537](https://network-pulse-staging-pr-963.herokuapp.com/profile/537)) shouldn't show you the 1st project header and CTA. 
 - This profile ([id:3](https://network-pulse-staging-pr-963.herokuapp.com/profile/3)) shouldn't show you the 2nd project header or the CTA.

File: components/profile-tab-group/profile-tab-group.jsx [hunk 1]
```diff
@@ -17,160 +13,106 @@ class ProfileTabGroup extends React.Component {
...
+    let visibleTabs = this.getAvailableTabs(props);
+    let activeTab;
+    let states = {
+      activeTab,
+      visibleTabs
+    };
 
-  componentWillReceiveProps(nextProps) {
-    this.fetchProfileEntries(nextProps.profileId);
+    if (visibleTabs.length === 0) return states;
```
Comment (alanmoo) [inline components/profile-tab-group/profile-tab-group.jsx]: I feel like this block could use some comments

File: components/profile-tab-group/profile-tab-group.jsx [hunk 2]
```diff
@@ -17,160 +13,106 @@ class ProfileTabGroup extends React.Component {
...
-
-    if (entries && entries.length > 0) {
-      content = <ProjectList entries={entries}
-        loadingData={false}
-        moreEntriesToFetch={false}
-        fetchData={()=>{}}
-        onModerationMode={false}
-      />;
+  renderTab() {
+    if (!this.state.activeTab) {
```
Comment (alanmoo) [inline components/profile-tab-group/profile-tab-group.jsx]: Comment plz

File: components/profile-tab-group/profile-tab-group.jsx [hunk 3]
```diff
@@ -17,160 +13,106 @@ class ProfileTabGroup extends React.Component {
...
-        header="Do you have something to share?"
-        linkComponent={<Link to={`/add`}>Add entry</Link>}
-      >
-        <p>If it might be useful to someone in our network, share it here.</p>
-      </HintMessage>;
-
-      if (this.state.entries.published.length === 0 && this.state.entries.created.length === 0) {
-        return this.renderProjects(``, [], prompt);
+  render() {
+    if (this.state.visibleTabs.length === 0) {
```
Comment (alanmoo) [inline components/profile-tab-group/profile-tab-group.jsx]: Comment

File: main.jsx [hunk 4]
```diff
@@ -75,7 +75,8 @@ const Routes = () => (
     <Route exact path="/help" render={() => <Redirect to="/latest"/>} />
     <Route path="/help/:helpType" component={Help} />
     <Route path="/moderation" component={Moderation} />
-    <Route path="/profile/:id" component={Profile} />
+    <Route exact path="/profile/:id" component={Profile} />
+    <Route path="/profile/:id/:tab" component={Profile} />
```
Comment (alanmoo) [inline main.jsx]: Doesn't React have a syntax for nesting in cases like this (would also be useful for `/help` above)?

Comment (mmmavis) [reply main.jsx]: If i remember correctly in v4 it doesn't support the old way of nesting routes. I spent quite some time few months ago trying to get it working close to the old way but didn't find any solution. I can file a followup ticket to tackle this... don't want this to be the blocker.

File: components/profile-project-tab/profile-project-tab.jsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  componentDidMount() {
+    this.fetchProfileEntries();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.tabName !== this.props.tabName) {
+      this.setState(this.getInitialState(nextProps), () => {
+        this.fetchProfileEntries(this.state.profileId);
```
[the remark this answers is not in the document]
Comment (mmmavis) [reply components/profile-project-tab/profile-project-tab.jsx]: good catch. this is old code to be removed

File: main.jsx (hunk 4 above)
Comment (alanmoo) [reply main.jsx]: That's fine especially since `help` is already like this, was just a thought as I read through.

**Comment by gvn** [comment]
Does this still need my review, or is @alanmoo doing it?

**Comment by mmmavis** [comment]
@gvn i just removed you from this PR. Alan is doing the code review already.

**Comment by xmatthewx** [comment]
rad! let's ship this today if possible. any blockers to that @mmmavis?

**Pull request merged**